service/service.go

```diff
@@ -430,6 +437,29 @@ func (s *service) BeforeServe(
 		} else {
 			log.Println("No management servers found.")
 		}
+
+		// Access the storagearrays key (which is a slice of maps)
+		storageArrays := secretParams.Get("storagearrays").([]interface{})
+		// Ensure there's at least one server and extract labels and parameters if any
+		if len(storageArrays) == 0 {
+			log.Println("No storage arrays found.")
+		} else {
+			// cycle through each storage array and extract Labels and Parameters maps
+			for _, storageArray := range storageArrays {
+				storageArrayMap := storageArray.(map[string]interface{})
+				if storageArrayMap["labels"] == nil {
+					storageArrayMap["labels"] = make(map[string]interface{})
+				}
+				if storageArrayMap["parameters"] == nil {
+					storageArrayMap["parameters"] = make(map[string]interface{})
+				}
+				storageArrayConfig := StorageArrayConfig{
+					Labels:     storageArrayMap["labels"].(map[string]interface{}),
+					Parameters: storageArrayMap["parameters"].(map[string]interface{}),
+				}
+				opts.StorageArrays = append(opts.StorageArrays, storageArrayConfig)
+			}
+		}
```

Comment:
If you extract this block into a utility function then you make unit testing easier. You can test a lot of use cases. The you can just do something like:

opts.StorageArrays = getStorageArrayInfo() or similar.

Reply:
done.